python/tank/descriptor/descriptor.py

```diff
@@ -371,6 +376,40 @@ def find_latest_version(self, constraint_pattern=None):
         latest._io_descriptor = self._io_descriptor.get_latest_version(constraint_pattern)
         return latest
 
+    def find_latest_cached_version(self, constraint_pattern=None):
```

Comment:
I renamed `has_remote()` to `has_remote_access()` and added it to the public API.